### PR TITLE
feature: implement virtual FileIO for Cloudflare

### PR DESCRIPTION
**Pull request by ssddOnTop**
Orignal issue: https://github.com/tailcallhq/tailcall/issues/750

- [x] Move worker logic from lib.rs to `worker.rs`

### CODE REVIEW

File: cloudflare/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,9 @@ async-trait = "0.1.77"
 reqwest-middleware = "0.2.4"
 reqwest = { version = "0.11", default-features = false }
 async-std = "1.12.0"
+wasm-logger = "0.2.0"
+log = "0.4.20"
+walkdir = "2.4.0"
```
Comment (tusharmath) [inline cloudflare/Cargo.toml:24]: where are we using this?

Comment (ssddOnTop) [reply cloudflare/Cargo.toml:24]: walkdir is not needed anymore, I switched to pathbuf. wasm-logger and log is optional if we need logging at all

File: cloudflare/src/handle.rs [hunk 2]
```diff
@@ -59,10 +58,19 @@ async fn init(env: Rc<worker::Env>) -> anyhow::Result<Arc<AppContext>> {
...
 async fn to_response(response: hyper::Response<hyper::Body>) -> anyhow::Result<worker::Response> {
   let buf = hyper::body::to_bytes(response).await?;
   let text = std::str::from_utf8(&buf)?;
-  Ok(worker::Response::ok(text).map_err(to_anyhow)?)
+  let mut response = worker::Response::ok(text).map_err(to_anyhow)?;
+  response
+    .headers_mut()
+    .append("Content-Type", "text/html")
+    .map_err(|e| anyhow!("{:?}", e))?;
+  Ok(response)
```
Comment (tusharmath) [inline cloudflare/src/handle.rs:73]: This should copy - header, body and status code from the hyper-response.

File: cloudflare/src/file.rs [hunk 3]
```diff
@@ -1,26 +1,92 @@
...
+    async_std::task::spawn_local(async move {
+      let mut vec = Vec::new();
+      // TODO: read files in parallel
+      for file in file_paths {
+        let content = io.read(&file).await.map_err(to_anyhow)?;
+        vec.push((content, file.to_string()));
+      }
+      anyhow::Ok(vec)
+    })
+    .await
```
Comment (tusharmath) [inline cloudflare/src/file.rs:84]: Let's make this run in parallel using `join_all`. and drop `spawn_local`

File: cloudflare/src/file.rs [hunk 4]
```diff
@@ -1,26 +1,92 @@
...
   }
 }
+
+/*async fn internal_shi(io: CloudflareFileIO, file_path: String, content: Vec<u8>) -> anyhow::Result<()> {
+  let r2 = R2Address::from_string(file_path)?;
+  io.put(&r2, content).await.map_err(to_anyhow)?;
+  Ok(())
+}*/
```
Comment (tusharmath) [inline cloudflare/src/file.rs]: delete commented code.

File: cloudflare/src/file.rs [hunk 5]
```diff
@@ -1,26 +1,86 @@
...
+
+  async fn read<'a>(&'a self, file_path: &'a str) -> anyhow::Result<String> {
+    let file_path = file_path.to_string();
+    let io = self.clone();
+    async_std::task::spawn_local(async move {
+      let r2 = R2Address::from_string(file_path)?;
+      let content = io.get(&r2).await.map_err(to_anyhow)?;
+      anyhow::Ok(content)
+    })
+    .await
```
Comment (tusharmath) [inline cloudflare/src/file.rs:69]: Drop `spawn_local`

**Pull request closed without merging**